src/@next/Select/Select.tsx

```diff
@@ -93,13 +94,33 @@ export const Select = ({
   };
 
   const handleFocus = () => {
+    if (options.length < 1) {
+      setPopoverActive(false);
+
+      return;
+    }
+
     setPopoverActive(true);
   };
 
   const handleSelectClick = () => {
     setPopoverActive(!popoverActive);
   };
 
+  useEffect(() => {
+    setMenuOptions(options);
+  }, [options]);
+
+  useEffect(() => {
+    if (inputValue != '' && options.length > 0) {
+      setPopoverActive(true);
+    }
+
+    if (inputValue === '' && options.length < 1) {
+      setPopoverActive(false);
+    }
+  }, [inputValue, options]);
```

Comment:
can we destruct the length from options here so we can use it in dependency array instead of the whole `options`?